### PR TITLE
EZP-31136: Added unlink to remove old link from ezurl table

### PR DESCRIPTION
**Pull request by mateuszdebinski**
| Question           | Answer
| ------------------ | ------------------
| **JIRA issue**     | [EZP-31136](https://jira.ez.no/browse/EZP-31136)
| **Bug/Improvement**| yes
| **New feature**    | no
| **Target version** | ``7.5` for bug fixes or improvements
| **BC breaks**      | no
| **Tests pass**     | yes
| **Doc needed**     | no

**TODO**:
- [x] Implement feature / fix a bug.
- [x] Implement tests.
- [x] Fix new code according to Coding Standards (`$ composer fix-cs`).
- [x] Ask for Code Review.

### CODE REVIEW

File: eZ/Publish/Core/FieldType/Url/UrlStorage.php [hunk 1]
```diff
@@ -62,6 +62,12 @@ public function storeFieldData(VersionInfo $versionInfo, Field $field, array $co
 
         $this->gateway->linkUrl($urlId, $field->id, $versionInfo->versionNo);
 
+        $this->gateway->unlinkUrl(
+            $field->id,
+            $versionInfo->versionNo,
+            [$urlId]
+        );
```
Comment (alongosz) [inline eZ/Publish/Core/FieldType/Url/UrlStorage.php:69]: This looks weird together. How does it work?

Comment (mateuszdebinski) [reply eZ/Publish/Core/FieldType/Url/UrlStorage.php:69]: This work like here: https://github.com/ezsystems/ezpublish-kernel/pull/2957
This is the same function to unlink old links from URL's tables

Comment (ViniTou) [reply eZ/Publish/Core/FieldType/Url/UrlStorage.php:69]: well, it is not to unlink old links but rather to unlink all but new ones, right?
@alongosz 
@mateuszdebinski

Comment (mateuszdebinski) [reply eZ/Publish/Core/FieldType/Url/UrlStorage.php:69]: The last parameter of the function indicates which urls should not be removed

**Comment by gggeek** [comment]
"As long as it is desired result that links that are used in old versions are no longer in link manager, because that will lead to it, right?"
I would say this seems the correct behaviour.
As long as re-publishing an old version _does_ lead to that url appearing again, of course ;-)

**Pull request merged**